quotefault/__init__.py

```diff
@@ -206,48 +206,43 @@ def get_quote_query(speaker: str = "", submitter: str = "", include_hidden: bool
 # display first 20 stored quotes
 @app.route('/storage', methods=['GET'])
 @auth.oidc_auth
-def get():
+def default_get():
+    return redirect("/storage/1")
+
+# display first 20 stored quotes
+@app.route('/storage/<page>', methods=['GET'])
+@auth.oidc_auth
+def get(page):
     """
     Show submitted quotes, only showing first 20 initially
     """
     metadata = get_metadata()
 
+    page = int(page)
+
+    query = get_quote_query(speaker = request.args.get('speaker'),
+        submitter = request.args.get('submitter'))
+        
+    rows = query.count()
+    rows = int(rows//20 + bool(rows%20 > 0))
+
+    if page > rows or page < 1:
+        return "Page Out of Bounds", 404
+
     # Get the most recent 20 quotes
-    quotes = get_quote_query(speaker = request.args.get('speaker'),
-        submitter = request.args.get('submitter')).limit(20).all()
+    quotes = query.offset((page-1)*20).limit(20).all()
 
     #tie any votes the user has made to their uid
     user_votes = Vote.query.filter(Vote.voter == metadata['uid']).all()
     return render_template(
         'bootstrap/storage.html',
         quotes=quotes,
         metadata=metadata,
-        user_votes=user_votes
-    )
-
-
-# display ALL stored quotes
-@app.route('/additional', methods=['GET'])
-@auth.oidc_auth
-def additional_quotes():
-    """
-    Show beyond the first 20 quotes
-    """
-
-    metadata = get_metadata()
-
-    # Get all the quotes
-    quotes = get_quote_query(speaker = request.args.get('speaker'),
-        submitter = request.args.get('submitter')).all()
-
-    #tie any votes the user has made to their uid
-    user_votes = db.session.query(Vote).filter(Vote.voter == metadata['uid']).all()
-
-    return render_template(
-        'bootstrap/additional_quotes.html',
-        quotes=quotes[20:],
-        metadata=metadata,
-        user_votes=user_votes
+        user_votes=user_votes,
+        page=page,
+        rows=rows,
+        begin=max(1, page-6),
+        end=min(page+6, rows) + 1

```

Comment:
Can this logic not be done in jinja? seems kinda weird to pass these when you're already passing the page.

Reply:
Min and Max are not jinja functions